core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -1320,8 +1320,30 @@ protected int getLineNumber() {
 		return lineNumber;
 	}
 
-	public StringBuilder getBuilder() {
+	private StringBuilder getBuilder() {
 		parsingBuilder.setLength(0);
 		return parsingBuilder;
 	}
+
+	/**
+	 * Appends the characters from codepoint into the string builder. This is the same
+	 * as Character#toChars but prevents the additional char array garbage for BMP codepoints.
+	 * @param dst the destination in which to append the characters
+	 * @param codePoint the codepoint to be appended
+	 */
+	private static void appendCodepoint(StringBuilder dst, int codePoint) {
+		/*
+		 * This code is copy pasted from Character#toChars but doesn't allocate new arrays.
```

Comment:
On the license issue: given that this is _such_ a basic/simple bit of code, I think that we can simply remove this specific comment to avoid confusion later. Other than that the PR (with the latest changes) looks good to me!